lightning/impl/sgd.py

```diff
@@ -395,6 +396,8 @@ def predict(self, X):
         try:
             assert_all_finite(self.coef_)
             pred = safe_sparse_dot(X, self.coef_.T)
+            if self.intercept_ is not None:
```

Comment:
Thanks a lot for the fix! `self.intercept_` is always set as long as `fit` is called and you already assume above that `self.coef_` is set. So I think we can remove the if above. What do you think?

Reply:
Thank you for your prompt review! I agree with you. Removed in 19ee3de216693fd267ff430fa6dbd610d31c72c0.